### DISCUSSION TITLE
Add command line control

### DISCUSSION THREAD

**Discussion by engelby** [File-New-Project/EarTrumpet#745] [Ideas]
I frequently switch between 2 different output sound devices.  Could there be a way to use Powershell to programmatically change between outputs?

**Comment by firewaran** [comment]
until this is considered or implemented i can recommend the option setdefaultsounddevice from [NirCmd](https://nircmd.nirsoft.net/setdefaultsounddevice.html)
I use that to switch between three setups: VR, Normal, Headphones

**Reply by dayeggpi** [reply]
EarTrumpet seems to be the only app in the market that :
1- applies the choice of output device per app, to Windows Volume Mixer settings
2- persists even if EarTrumpet is closed (thanks to point 1)

Which makes it right now the best app in the market (on top of all other perks).

Being able to trigger these choices via command line would be amazing as it would allow for programmatic changes...
example of a command line would be :
`EarTrumpet.exe --set vlc.exe "CABLE-C Input (VB-Audio Cable C)"`

I guess it would need two additional command lines (to not touch the GUI and separate that feature for cli users only) :
1- to list out the apps filename that can be used with the set command (e.g. indeed, "VLC Media Player" as displayed in EarTrumpet GUI, would be vlc.exe, this would solve for other apps that have variable "title" like Chrome if playing a video etc.). This would be obtained via another command (e.g. `EarTrumpet.exe --list-apps`)
2- to list out the device's list (e.g. "CABLE-C Input (VB-Audio Cable C)"). This would be obtained via another command (e.g. `EarTrumpet.exe --list-devices`)

Note: issue with NirSoft tools is that it either doesnt change Windows Mixer settings properly, or doesn't allow per app control. EarTrumpt seems the only way to go, but it's missing that CLI aspect.

**Reply by dayeggpi** [reply]
@riverar I ended up implementing something to solve for it. It will be my first PR on github, so i might not be great in the process (learning curve) but I see from the CONTRIBUTING.md that I need an approval

**Comment by engelby** [comment]
Thanks!  Somehow my google-fu didn't find that utility.   For those who may be wondering, I can confirm this works under Windows 10.
